doc/.gitignore

```diff
@@ -1 +1 @@
-book
```

Comment:
I think we still need `book/` in the `.gitignore` because when we do `mdbook build` locally, it'll output it in `book/`.